detection_rules/utils.py

```diff
@@ -306,6 +307,14 @@ def clear_caches():
     _cache.clear()
 
 
+def rulename_to_filename(name: str, tactic_name: str = None) -> str:
+    name = re.sub(r'[^_a-z0-9]+', '_', name.strip().lower()).strip('_') + '.toml'
+    if tactic_name:
+        pre = re.sub(r'[^_a-z0-9]+', '_', tactic_name.strip().lower()).strip('_')
+        name = f'{pre}_{name}'
+    return name
```

Comment:
We should probably clean this update an deduplicate with an inner function or lambda.